libs/protocol-ui/src/components/molecules/ControlledDatePicker/ControlledDatePickerButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {
+  Button,
+  ButtonProps,
+  forwardRef,
+  ComponentWithAs,
+} from '@chakra-ui/react';
+import { HiOutlineCalendar } from 'react-icons/hi';
+
+export type ControlledDatePickerButtonProps = ButtonProps & {
+  value?: React.ReactNode;
+  onClick?: () => void;
+  ref?: HTMLInputElement;
+  buttonDisabled?: boolean;
+};
+
+export type CustomDatePickerButton = ComponentWithAs<
+  'button',
+  ControlledDatePickerButtonProps
+> & {
+  Button: typeof Button;
+};
```

Comment:
Is this used anywhere else?

Reply:
It's used in both the custom buttons for the Date Picker -- without this extension of the `ComponentWithAs` Chakra has trouble picking up the types

Reply:
Ah okay my bad